### DISCUSSION TITLE
Text Formatting - Code Block

### DISCUSSION THREAD

**Discussion by kasey-seo** [twilio-labs/paste#2135] [Q&A] [answered]
Hi team,

I need to design a modified version of [this page](https://www.twilio.com/docs/taskrouter/limits) in the console just wondering if there is a way to stylise some part of a paragraph like a code or code block? 

<img width="347" alt="Screenshot 2022-01-12 at 14 18 17" src="https://user-images.githubusercontent.com/94976467/149351839-44bf5bc0-6f73-4d02-b0a3-cbaf7cdbbe89.png">

**Comment by kate-porter** [comment, marked as answer]
Hey @kasey-seo - there is no Paste component or pattern around this yet, but you can still achieve this look using Box and Text. In fact, we do it quite frequently on the Paste docs site ([here's an example](https://paste.twilio.design/components/alert#accessibility)).

I noticed your original post in Slack also asked about how to do this in Figma. Unfortunately, the answer is that it'll have to be a little hacky—there's not a great way to style text inline when it's got a custom background color, border radius, etc. What'll you basically need to do is style the code snippets separately from the paragraph text, and then drop the code snippets on top of the paragraph text to create the effect. [Here's a quick vid](https://www.loom.com/share/7a8567988c0745e1a627230162a4a329) that explains what I mean.

I hope that helps! Let us know if you have any other questions on this.

**Reply by kasey-seo** [reply]
Thanks Katie, the video is very helpful :)

**Reply by bhamtil-twilio** [reply]
Hello! To piggy back off this question – is there a component in the works for a large code sample block (not in-line with text)? I've seen a handful of use cases across the platform and they all appear to be one-offs. Examples: Ahoy workflows, Request Inspector in Error Log details

**Reply by shleewhite** [reply]
Hey @bhamtil-twilio,

We do have a Code Block component in our backlog, but it hasn't been prioritized. For now, if you don't need syntax highlighting or theming, our recommendation is to use the Box component and the `'fontFamilyCode'`. Here is a live example: https://codesandbox.io/s/simple-code-block-yr0rgd?file=/src/App.tsx

If you have any examples of code blocks in products, can you add them to [the mood board](https://www.figma.com/file/6MlUZI1JHp8aIWE5JCGlm0/Paste-Moodboard?node-id=1510%3A66643)?

**Reply by kate-porter** [reply]
Hey @bhamtil-twilio - just wanted to followup here that we've actually pulled Code Block up in priority and will be releasing it as a component in Paste this sprint!

**Reply by bhamtil-twilio** [reply]
Woohoo, that's great news! Thanks!

**Reply by bhamtil-twilio** [reply]
Hi @shleewhite and @katieporter, does the Code Block component automatically format and color the code content inside it? Or are we meant to do that ourselves? From what I understand current code blocks use a third-party tool to "prettify" the code inside it automatically.

**Reply by SiTaggart** [reply]
For blocks of code, we will likely draw the line at syntax highlighting the code that is provided to us. 

We generally don't touch the content that is passed to a component, because it's not ours to touch. We don't know and can't infer what you want it to be, so it's best to leave that up to the author to give you the full flexibility and control over your content.

**Reply by arigsigal** [reply]
Incredibly excited that this is being prioritized! Code snippets feature heavily in our onboarding experiences but today don't support syntax hilighting or long line wrapping (see screenshot). One thing that would be really useful for us for cURL snippets is the ability to toggle to display Auth Token. Is that something you'd consider adding to the roadmap? 
![image (15)](https://user-images.githubusercontent.com/13924309/183529312-ccfd0806-0cbd-4d4a-bc71-a82b044337e5.png)

**Comment by SiTaggart** [comment]
[Code block](https://paste.twilio.design/components/code-block) and [inline Code](https://paste.twilio.design/components/inline-code) are now available in Paste.
